apps/hash-frontend/src/pages/notes.page/create-quick-note.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Entity,
+  EntityRootType,
+  OwnedById,
+  Subgraph,
+} from "@local/hash-subgraph";
+import { Box, Skeleton } from "@mui/material";
+import {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useKeys } from "rooks";
+
+import { useBlockProtocolGetEntity } from "../../components/hooks/block-protocol-functions/knowledge/use-block-protocol-get-entity";
+import { ArrowTurnDownLeftRegularIcon } from "../../shared/icons/arrow-turn-down-left-regular-icon";
+import { Button } from "../../shared/ui";
+import { useAuthenticatedUser } from "../shared/auth-info-context";
+import { useCreateBlockCollection } from "../shared/use-create-block-collection";
+import { EditableQuickNote } from "./editable-quick-note";
+
+export const CreateQuickNote: FunctionComponent<{
+  initialQuickNoteEntity?: Entity | null;
+  initialQuickNoteEntitySubgraph?: Subgraph<EntityRootType>;
+  refetchQuickNotes: () => Promise<void>;
+  onCreatingQuickNote: (quickNoteEntity: Entity) => void;
+}> = ({
+  initialQuickNoteEntity,
+  initialQuickNoteEntitySubgraph,
+  onCreatingQuickNote,
+  refetchQuickNotes,
+}) => {
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const { authenticatedUser } = useAuthenticatedUser();
+
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    const handleFocus = (event: FocusEvent) =>
+      setIsFocused(
+        !!wrapperRef.current &&
+          event.target instanceof Node &&
+          wrapperRef.current.contains(event.target),
+      );
+
+    const handleBlur = () => setIsFocused(false);
+
+    document.addEventListener("focus", handleFocus, true);
+    document.addEventListener("blur", handleBlur, true);
+
+    return () => {
+      document.removeEventListener("focus", handleFocus, true);
+      document.removeEventListener("blur", handleBlur, true);
+    };
+  }, []);
+
+  const [creatingNewQuickNote, setCreatingNewQuickNote] = useState(false);
+
+  const [quickNoteEntity, setQuickNoteEntity] = useState<Entity>();
+
+  const [quickNoteSubgraph, setQuickNoteSubgraph] =
+    useState<Subgraph<EntityRootType>>();
+
+  if (!quickNoteSubgraph && initialQuickNoteEntitySubgraph) {
+    setQuickNoteSubgraph(initialQuickNoteEntitySubgraph);
+  }
+
+  const { createBlockCollectionEntity } = useCreateBlockCollection({
+    ownedById: authenticatedUser.accountId as OwnedById,
+  });
+  const { getEntity } = useBlockProtocolGetEntity();
+
+  const createQuickNote = useCallback(async () => {
+    const createdQuickNoteEntity = await createBlockCollectionEntity({
+      kind: "quickNote",
+    });
+
+    const { data } = await getEntity({
+      data: {
+        entityId: createdQuickNoteEntity.metadata.recordId.entityId,
+        graphResolveDepths: {
+          hasLeftEntity: { incoming: 2, outgoing: 2 },
+          hasRightEntity: { incoming: 2, outgoing: 2 },
```

Comment:
Can we add a comment expalining what entities it needs these for? When we can be more selectively in what data we fetch it'll help

Reply:
Sure, it's to get the blocks and the block text data. Have added in https://github.com/hashintel/hash/pull/3434/commits/297ccde87f2f3666fcdf922b8425b13b80c5e18e